tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/const-pattern.bal

```diff
@@ -688,6 +688,68 @@ function constPatternWithPredeclaredPrefix2(int[] x) returns int {
     }
 }
 
+function testMatchStmtInsideForeachString1() {
+    string input = "1x2";
+    string output = "";
+    foreach string char in input {
+        match char {
+            "1" => {
+                output = output.concat("One");
+            }
+            "2" => {
+                output = output.concat("Two");
+            }
+            _ => {
+                output = output.concat("Other");
+            }
+        }
+    }
+    assertEquals("OneOtherTwo", output);
+}
+
+type CHAR string;
+
+function testMatchStmtInsideForeachString2() {
+    CHAR input = "1x2";
+    string output = "";
+    foreach string char in input {
+        match char {
+            "1" => {
+                output = output.concat("One");
+            }
+            "2" => {
+                output = output.concat("Two");
+            }
+            _ => {
+                output = output.concat("Other");
+            }
+        }
+    }
+    assertEquals("OneOtherTwo", output);
+}
+
+function testMatchStmtInsideForeachInt() {
+    int:Signed8[] input = [7, 2, 2];
+    string output = "";
+    foreach int number in input {
+        match number {
+            2 => {
+                output = output.concat("Two");
+            }
+            _ => {
+                output = output.concat("Other");
+            }
+        }
+    }
+    assertEquals("OtherTwoTwo", output);
+}
+
+function testMatchStmtInsideForeach() {
+    testMatchStmtInsideForeachString1();
+    testMatchStmtInsideForeachString2();
+    testMatchStmtInsideForeachInt();
+}
+
```

Comment:
Shall we add one test with constant variables.
ex:- 
```
function testMatchStmtInsideForeachString1() {
    string input = "1x2";
    string output = "";
    const string one = "1";
    const string two = "2";
    
    foreach string char in input {
        match char {
            one => {
                output = output.concat("One");
            }
            two => {
                output = output.concat("Two");
            }
            _ => {
                output = output.concat("Other");
            }
        }
    }
    assertEquals("OneOtherTwo", output);
}
```

Reply:
will add

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/37334/commits/a520b494cfd7b6ba1c35380a6dfa2ee7ca03ba01